roles/storage-glusterfs/tasks/provision.yml

```diff
@@ -17,8 +17,32 @@
 
 - set_fact:
     heketi_url: "{{ route_cmd.stdout }}"
-  when: route_cmd is defined
+  when:
+    - cluster == 'openshift'
+    - route_cmd is defined
+
+- name: Determine the heketi route for k8s, IP
+  command: "kubectl get pod -n kube-system -l heketi=pod -o jsonpath='{.items[0].status.podIP}'"
+  register: heketi_pod_ip
+  when:
+    - cluster == 'k8s'
+    - heketi_url is not defined
 
+- name: Determine the heketi route for k8s, port
+  command: "kubectl get pod -n kube-system -l heketi=pod -o jsonpath='{.items[0].spec.containers[0].ports[0].containerPort}'"
```

Comment:
why do lines 25 and 32 have `kube-system` namespaces and other tasks in this playbook have it as `{{ glusterfs_namespace }}`?

Reply:
@alexxa good catch, will fix